<?xml version="1.0" encoding="UTF-8" ?>
<rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom" xmlns:dc="http://purl.org/dc/elements/1.1/">
  <channel>
    <title>An error occurs Redash Docker on startup</title>
    <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608</link>
    <description>Hi,

When I started Redash docker-compose up:

```bash
PostgreSQL Database directory appears to contain a database; Skipping initialization
redash-master-postgres-1   | 
redash-master-postgres-1   | LOG:  database system was shut down at 2022-06-09 14:29:51 UTC
redash-master-postgres-1   | LOG:  MultiXact member wraparound protections are now enabled
redash-master-postgres-1   | LOG:  database system is ready to accept connections
redash-master-postgres-1   | LOG:  autovacuum launcher started
redash-master-email-1      | MailDev webapp running at http://0.0.0.0:80
redash-master-email-1      | MailDev SMTP Server running at 0.0.0.0:25
redash-master-worker-1     | Starting dev RQ worker...
redash-master-scheduler-1  | Starting dev RQ scheduler...
redash-master-server-1     | Traceback (most recent call last):
redash-master-server-1     |   File &quot;/app/manage.py&quot;, line 6, in &lt;module&gt;
redash-master-server-1     |     from redash.cli import manager
redash-master-server-1     |   File &quot;/app/redash/__init__.py&quot;, line 56, in &lt;module&gt;
redash-master-server-1     |     import_query_runners(settings.QUERY_RUNNERS)
redash-master-server-1     |   File &quot;/app/redash/query_runner/__init__.py&quot;, line 438, in import_query_runners
redash-master-server-1     |     __import__(runner_import)
redash-master-server-1     |   File &quot;/app/redash/query_runner/big_query.py&quot;, line 17, in &lt;module&gt;
redash-master-server-1     |     import apiclient.errors
redash-master-server-1     |   File &quot;/usr/local/lib/python3.7/site-packages/apiclient/__init__.py&quot;, line 8, in &lt;module&gt;
redash-master-server-1     |     from googleapiclient import discovery
redash-master-server-1     |   File &quot;/usr/local/lib/python3.7/site-packages/googleapiclient/discovery.py&quot;, line 56, in &lt;module&gt;
redash-master-server-1     |     from googleapiclient import _auth
redash-master-server-1     |   File &quot;/usr/local/lib/python3.7/site-packages/googleapiclient/_auth.py&quot;, line 33, in &lt;module&gt;
redash-master-server-1     |     import oauth2client.client
redash-master-server-1     |   File &quot;/usr/local/lib/python3.7/site-packages/oauth2client/client.py&quot;, line 45, in &lt;module&gt;
redash-master-server-1     |     from oauth2client import crypt
redash-master-server-1     |   File &quot;/usr/local/lib/python3.7/site-packages/oauth2client/crypt.py&quot;, line 55, in &lt;module&gt;
redash-master-server-1     |     from oauth2client import _pycrypto_crypt
redash-master-server-1     |   File &quot;/usr/local/lib/python3.7/site-packages/oauth2client/_pycrypto_crypt.py&quot;, line 17, in &lt;module&gt;
redash-master-server-1     |     from Crypto.PublicKey import RSA
redash-master-server-1     |   File &quot;/usr/local/lib/python3.7/site-packages/Crypto/PublicKey/RSA.py&quot;, line 585
redash-master-server-1     |     except ValueError, IndexError:
redash-master-server-1     |                      ^
redash-master-server-1     | SyntaxError: invalid syntax
redash-master-server-1 exited with code 1
```

Please help!!!!</description>
    
    <lastBuildDate>Tue, 21 Jun 2022 16:10:20 +0000</lastBuildDate>
    <category>Support</category>
    <atom:link href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss" rel="self" type="application/rss+xml" />
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[jesse]]></dc:creator>
        <description><![CDATA[
            <p>This is very strange. Especially since the the CLI command has been broken for awhile…</p>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/27">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/27</link>
        <pubDate>Tue, 21 Jun 2022 16:10:20 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-27</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[dishka_krauch]]></dc:creator>
        <description><![CDATA[
            <p>Really strage error cause I can invite people with shell command but cannot do it with web page.</p>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/26">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/26</link>
        <pubDate>Mon, 20 Jun 2022 21:43:03 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-26</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[dishka_krauch]]></dc:creator>
        <description><![CDATA[
            <p>We have cured our .env file and finally test message was sent successfully.<br>
Now we’re trying to understand what is wrong with env variable SERVER_NAME cause after invintation any new memer this message pops up:</p>
<pre><code class="lang-auto">RuntimeError: Application was not able to create a URL adapter for request independent URL generation. You might be able to fix this by setting the SERVER_NAME config variable.
</code></pre>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/25">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/25</link>
        <pubDate>Mon, 20 Jun 2022 21:31:46 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-25</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[dishka_krauch]]></dc:creator>
        <description><![CDATA[
            <p>I think you were right about my own host issues. All issues was related to corporate firewall.</p>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/23">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/23</link>
        <pubDate>Fri, 17 Jun 2022 14:07:43 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-23</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[jesse]]></dc:creator>
        <description><![CDATA[
            <p>Glad to hear it. What was the issue with pycrypto?</p>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/22">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/22</link>
        <pubDate>Fri, 17 Jun 2022 14:06:06 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-22</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[dishka_krauch]]></dc:creator>
        <description><![CDATA[
            <p>Yeah, I see, thank you for quick replies. It was really helpfull.<br>
Finally service starts and I can reach it.<br>
Now we have some troubles with corporate smtp server but it’s easy to solve.</p>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/21">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/21</link>
        <pubDate>Fri, 17 Jun 2022 14:05:35 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-21</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[jesse]]></dc:creator>
        <description><![CDATA[
            <p>That error means you didn’t build the front-end assets. To do this you need to run <code>yarn build</code>. Then try again and it should load.</p>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/20">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/20</link>
        <pubDate>Thu, 16 Jun 2022 20:06:44 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-20</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[dishka_krauch]]></dc:creator>
        <description><![CDATA[
            <p>Finally I switched ports and can reach redash web setup page. It failes with other errors related to yarn as I can guess.</p>
<pre><code class="lang-auto">FileNotFoundError: [Errno 2] No such file or directory: '/app/redash/settings/../../client/dist/index.html'
</code></pre>
<p>Looks like it had been solving by you a years go. Give me some time to check evertying out.</p>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/19">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/19</link>
        <pubDate>Thu, 16 Jun 2022 20:05:12 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-19</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[jesse]]></dc:creator>
        <description><![CDATA[
            <aside class="quote group-Team" data-username="jesse" data-post="16" data-topic="10608">
<div class="title">
<div class="quote-controls"></div>
<img loading="lazy" alt="" width="20" height="20" src="https://sjc6.discourse-cdn.com/standard17/user_avatar/discuss.redash.io/jesse/40/3428_2.png" class="avatar"> jesse:</div>
<blockquote>
<p>It looks like you were not shown an onboarding screen to create an organisation.</p>
</blockquote>
</aside>
<p>Can you confirm whether anything has loaded in your browser yet?</p>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/18">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/18</link>
        <pubDate>Thu, 16 Jun 2022 19:59:59 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-18</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[dishka_krauch]]></dc:creator>
        <description><![CDATA[
            <p>Okay, I got your pretty straight answer. I’ll try to make it on my own.</p>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/17">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/17</link>
        <pubDate>Thu, 16 Jun 2022 19:59:26 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-17</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[jesse]]></dc:creator>
        <description><![CDATA[
            <p>Boy you are just plagued with odd issues I’ve never seen before. It looks like you were not shown an onboarding screen to create an organisation. Does anything load in your browser?</p>
<p>What <em>exact</em> steps have you followed so far? It’s not enough to point to the doc page because collectively these instructions are followed dozens-hundreds of times per month and don’t produce these issues. I believe there must be an environment issue on your box, potentially related to Centos but it’s hard to say.</p>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/16">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/16</link>
        <pubDate>Thu, 16 Jun 2022 19:44:33 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-16</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[dishka_krauch]]></dc:creator>
        <description><![CDATA[
            <p>It’s strange cause I had to run this command twice to pass this error.<br>
So for now I have following output after docker-compose up command.</p>
<pre><code class="lang-auto">[+] Running 6/6
 ⠿ Container redash-master-email-1      Created                                                                                                                                                           0.0s
 ⠿ Container redash-master-redis-1      Running                                                                                                                                                           0.0s
 ⠿ Container redash-master-postgres-1   Running                                                                                                                                                           0.0s
 ⠿ Container redash-master-server-1     Created                                                                                                                                                           0.0s
 ⠿ Container redash-master-worker-1     Created                                                                                                                                                           0.0s
 ⠿ Container redash-master-scheduler-1  Created                                                                                                                                                           0.0s
Attaching to redash-master-email-1, redash-master-postgres-1, redash-master-redis-1, redash-master-scheduler-1, redash-master-server-1, redash-master-worker-1
redash-master-email-1      | MailDev webapp running at http://0.0.0.0:80
redash-master-email-1      | MailDev SMTP Server running at 0.0.0.0:25
redash-master-scheduler-1  | Starting dev RQ scheduler...
redash-master-worker-1     | Starting dev RQ worker...
redash-master-server-1     |  * Environment: production
redash-master-server-1     |    WARNING: This is a development server. Do not use it in a production deployment.
redash-master-server-1     |    Use a production WSGI server instead.
redash-master-server-1     |  * Debug mode: on
redash-master-server-1     | [2022-06-16 19:34:55,341][PID:1][INFO][werkzeug]  * Running on http://0.0.0.0:5000/ (Press CTRL+C to quit)
redash-master-server-1     | [2022-06-16 19:34:55,382][PID:1][INFO][werkzeug]  * Restarting with inotify reloader
redash-master-worker-1     | [2022-06-16 19:34:56,163][PID:8][INFO][rq.worker] Worker rq:worker:7c49c19d7f9d450686848352ce2f7bbe: started, version 1.5.0
redash-master-worker-1     | [2022-06-16 19:34:56,164][PID:8][INFO][rq.worker] *** Listening on periodic, emails, default, scheduled_queries, queries, schemas...
redash-master-worker-1     | [2022-06-16 19:34:56,164][PID:8][INFO][rq.worker] Cleaning registries for queue: periodic
redash-master-worker-1     | [2022-06-16 19:34:56,165][PID:8][INFO][rq.worker] Cleaning registries for queue: emails
redash-master-worker-1     | [2022-06-16 19:34:56,166][PID:8][INFO][rq.worker] Cleaning registries for queue: default
redash-master-worker-1     | [2022-06-16 19:34:56,167][PID:8][INFO][rq.worker] Cleaning registries for queue: scheduled_queries
redash-master-worker-1     | [2022-06-16 19:34:56,168][PID:8][INFO][rq.worker] Cleaning registries for queue: queries
redash-master-worker-1     | [2022-06-16 19:34:56,169][PID:8][INFO][rq.worker] Cleaning registries for queue: schemas
redash-master-scheduler-1  | [2022-06-16 19:34:56,196][PID:8][INFO][redash.tasks.schedule] Scheduling 305c0cae0a196ae96915fd2b6f81001c435aad65 (refresh_queries) with interval 30.
redash-master-scheduler-1  | [2022-06-16 19:34:56,197][PID:8][INFO][redash.tasks.schedule] Scheduling 59a84668e68687338646f735965213c58e814b32 (remove_ghost_locks) with interval 60.
redash-master-scheduler-1  | [2022-06-16 19:34:56,198][PID:8][INFO][redash.tasks.schedule] Scheduling 75540cdf868e7873f5eec072177423b13b98dece (empty_schedules) with interval 3600.
redash-master-scheduler-1  | [2022-06-16 19:34:56,199][PID:8][INFO][redash.tasks.schedule] Scheduling 52bcd40c254552539398db3cdb15055d4a9a536a (refresh_schemas) with interval 1800.
redash-master-scheduler-1  | [2022-06-16 19:34:56,200][PID:8][INFO][redash.tasks.schedule] Scheduling 18281a865ed3d1a60f366aeb8596fe2283aa421f (sync_user_details) with interval 60.
redash-master-scheduler-1  | [2022-06-16 19:34:56,200][PID:8][INFO][redash.tasks.schedule] Scheduling 797f4f959c6c96c738efd445b17db75742db80a9 (send_aggregated_errors) with interval 3600.
redash-master-scheduler-1  | [2022-06-16 19:34:56,201][PID:8][INFO][redash.tasks.schedule] Scheduling 526f2c7f55dfe92457914cd7df02cfffe4dec877 (version_check) with interval 86400.
redash-master-scheduler-1  | [2022-06-16 19:34:56,202][PID:8][INFO][redash.tasks.schedule] Scheduling e27209059575fcc17c527c47d0957cb21756e551 (cleanup_query_results) with interval 300.
redash-master-scheduler-1  | [2022-06-16 19:34:56,203][PID:8][INFO][rq_scheduler.scheduler] Registering birth
redash-master-worker-1     | [2022-06-16 19:34:56,207][PID:8][INFO][rq.worker] periodic: 18281a865ed3d1a60f366aeb8596fe2283aa421f
redash-master-worker-1     | [2022-06-16 19:34:56,227][PID:47][INFO][rq.worker] periodic: Job OK (18281a865ed3d1a60f366aeb8596fe2283aa421f)
redash-master-worker-1     | [2022-06-16 19:34:56,228][PID:47][INFO][rq.worker] Result is kept for 600 seconds
redash-master-worker-1     | [2022-06-16 19:34:56,237][PID:8][INFO][rq.worker] periodic: 305c0cae0a196ae96915fd2b6f81001c435aad65
redash-master-worker-1     | [2022-06-16 19:34:56,247][PID:48][INFO][rq.job.redash.tasks.queries.maintenance] job.func_name=redash.tasks.queries.maintenance.refresh_queries job.id=305c0cae0a196ae96915fd2b6f81001c435aad65 Refreshing queries...
redash-master-worker-1     | [2022-06-16 19:34:56,283][PID:48][INFO][rq.job.redash.tasks.queries.maintenance] job.func_name=redash.tasks.queries.maintenance.refresh_queries job.id=305c0cae0a196ae96915fd2b6f81001c435aad65 Done refreshing queries: {'outdated_queries_count': 0, 'last_refresh_at': 1655408096.2826107, 'query_ids': '[]'}
redash-master-worker-1     | [2022-06-16 19:34:56,286][PID:48][INFO][rq.worker] periodic: Job OK (305c0cae0a196ae96915fd2b6f81001c435aad65)
redash-master-worker-1     | [2022-06-16 19:34:56,286][PID:48][INFO][rq.worker] Result is kept for 600 seconds
redash-master-postgres-1   | LOG:  unexpected EOF on client connection with an open transaction
redash-master-worker-1     | [2022-06-16 19:34:56,300][PID:8][INFO][rq.worker] periodic: 526f2c7f55dfe92457914cd7df02cfffe4dec877
redash-master-worker-1     | [2022-06-16 19:34:56,314][PID:49][INFO][root] Performing version check.
redash-master-worker-1     | [2022-06-16 19:34:56,314][PID:49][INFO][root] Current version: 11.0.0-dev
redash-master-worker-1     | [2022-06-16 19:34:56,347][PID:49][ERROR][rq.worker] AttributeError: 'NoneType' object has no attribute 'get_setting'
redash-master-worker-1     | Traceback (most recent call last):
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/rq/worker.py", line 934, in perform_job
redash-master-worker-1     |     rv = job.perform()
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/rq/job.py", line 686, in perform
redash-master-worker-1     |     self._result = self._execute()
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/rq/job.py", line 692, in _execute
redash-master-worker-1     |     return self.func(*self.args, **self.kwargs)
redash-master-worker-1     |   File "/app/redash/tasks/general.py", line 35, in version_check
redash-master-worker-1     |     run_version_check()
redash-master-worker-1     |   File "/app/redash/version_check.py", line 75, in run_version_check
redash-master-worker-1     |     if Organization.query.first().get_setting("beacon_consent"):
redash-master-worker-1     | AttributeError: 'NoneType' object has no attribute 'get_setting'
redash-master-worker-1     | Traceback (most recent call last):
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/rq/worker.py", line 934, in perform_job
redash-master-worker-1     |     rv = job.perform()
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/rq/job.py", line 686, in perform
redash-master-worker-1     |     self._result = self._execute()
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/rq/job.py", line 692, in _execute
redash-master-worker-1     |     return self.func(*self.args, **self.kwargs)
redash-master-worker-1     |   File "/app/redash/tasks/general.py", line 35, in version_check
redash-master-worker-1     |     run_version_check()
redash-master-worker-1     |   File "/app/redash/version_check.py", line 75, in run_version_check
redash-master-worker-1     |     if Organization.query.first().get_setting("beacon_consent"):
redash-master-worker-1     | AttributeError: 'NoneType' object has no attribute 'get_setting'
redash-master-postgres-1   | LOG:  unexpected EOF on client connection with an open transaction
</code></pre>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/15">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/15</link>
        <pubDate>Thu, 16 Jun 2022 19:40:17 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-15</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[jesse]]></dc:creator>
        <description><![CDATA[
            <p>Looks like you didn’t set up your metadata database.</p>
<p><code>docker-compose run --rm server create_db</code></p>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/14">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/14</link>
        <pubDate>Thu, 16 Jun 2022 16:08:34 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-14</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[dishka_krauch]]></dc:creator>
        <description><![CDATA[
            <p>Here is new error after docker-compose up command:</p>
<pre><code class="lang-auto">redash-master-postgres-1   | ERROR:  relation "query_results" does not exist at character 13
redash-master-postgres-1   | STATEMENT:  DELETE FROM query_results WHERE query_results.id IN (SELECT query_results.id
redash-master-postgres-1   |    FROM query_results LEFT OUTER JOIN queries ON query_results.id = queries.latest_query_data_id
redash-master-postgres-1   |    WHERE queries.id IS NULL AND query_results.retrieved_at &lt; '2022-06-09T14:53:40.363540'::timestamp
redash-master-postgres-1   |     LIMIT 100)
redash-master-worker-1     | [2022-06-16 14:53:40,399][PID:54][ERROR][rq.worker] sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedTable) relation "query_results" does not exist
redash-master-worker-1     | LINE 1: DELETE FROM query_results WHERE query_results.id IN (SELECT ...
redash-master-worker-1     |                     ^
redash-master-worker-1     |
redash-master-worker-1     | [SQL: DELETE FROM query_results WHERE query_results.id IN (SELECT query_results.id
redash-master-worker-1     | FROM query_results LEFT OUTER JOIN queries ON query_results.id = queries.latest_query_data_id
redash-master-worker-1     | WHERE queries.id IS NULL AND query_results.retrieved_at &lt; %(retrieved_at_1)s
redash-master-worker-1     |  LIMIT %(param_1)s)]
redash-master-worker-1     | [parameters: {'retrieved_at_1': datetime.datetime(2022, 6, 9, 14, 53, 40, 363540), 'param_1': 100}]
redash-master-worker-1     | (Background on this error at: http://sqlalche.me/e/f405)
redash-master-worker-1     | Traceback (most recent call last):
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1249, in _execute_context
redash-master-worker-1     |     cursor, statement, parameters, context
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 580, in do_execute
redash-master-worker-1     |     cursor.execute(statement, parameters)
redash-master-worker-1     | psycopg2.errors.UndefinedTable: relation "query_results" does not exist
redash-master-worker-1     | LINE 1: DELETE FROM query_results WHERE query_results.id IN (SELECT ...
redash-master-worker-1     |                     ^
redash-master-worker-1     |
redash-master-worker-1     |
redash-master-worker-1     | The above exception was the direct cause of the following exception:
redash-master-worker-1     |
redash-master-worker-1     | Traceback (most recent call last):
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/rq/worker.py", line 934, in perform_job
redash-master-worker-1     |     rv = job.perform()
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/rq/job.py", line 686, in perform
redash-master-worker-1     |     self._result = self._execute()
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/rq/job.py", line 692, in _execute
redash-master-worker-1     |     return self.func(*self.args, **self.kwargs)
redash-master-worker-1     |   File "/app/redash/tasks/queries/maintenance.py", line 142, in cleanup_query_results
redash-master-worker-1     |     ).delete(synchronize_session=False)
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/query.py", line 3727, in delete
redash-master-worker-1     |     delete_op.exec_()
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/persistence.py", line 1693, in exec_
redash-master-worker-1     |     self._do_exec()
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/persistence.py", line 1921, in _do_exec
redash-master-worker-1     |     self._execute_stmt(delete_stmt)
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/persistence.py", line 1698, in _execute_stmt
redash-master-worker-1     |     self.result = self.query._execute_crud(stmt, self.mapper)
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/query.py", line 3375, in _execute_crud
redash-master-worker-1     |     return conn.execute(stmt, self._params)
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 988, in execute
redash-master-worker-1     |     return meth(self, multiparams, params)
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/sql/elements.py", line 287, in _execute_on_connection
redash-master-worker-1     |     return connection._execute_clauseelement(self, multiparams, params)
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1107, in _execute_clauseelement
redash-master-worker-1     |     distilled_params,
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1253, in _execute_context
redash-master-worker-1     |     e, statement, parameters, cursor, context
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1473, in _handle_dbapi_exception
redash-master-worker-1     |     util.raise_from_cause(sqlalchemy_exception, exc_info)
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 398, in raise_from_cause
redash-master-worker-1     |     reraise(type(exception), exception, tb=exc_tb, cause=cause)
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 152, in reraise
redash-master-worker-1     |     raise value.with_traceback(tb)
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1249, in _execute_context
redash-master-worker-1     |     cursor, statement, parameters, context
redash-master-worker-1     |   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 580, in do_execute
redash-master-worker-1     |     cursor.execute(statement, parameters)
redash-master-worker-1     | sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedTable) relation "query_results" does not exist
redash-master-worker-1     | LINE 1: DELETE FROM query_results WHERE query_results.id IN (SELECT ...

</code></pre>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/13">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/13</link>
        <pubDate>Thu, 16 Jun 2022 14:54:59 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-13</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[dishka_krauch]]></dc:creator>
        <description><![CDATA[
            <p>I’ve restarted everything from the scratch and error dessapeared as it always happens… Right now can’t reach redash master web pache after port changing from 5000 to 8088.</p>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/12">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/12</link>
        <pubDate>Thu, 16 Jun 2022 14:53:07 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-12</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[dishka_krauch]]></dc:creator>
        <description><![CDATA[
            <p>Got it. Give me some time to check it out.</p>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/11">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/11</link>
        <pubDate>Thu, 16 Jun 2022 14:22:13 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-11</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[jesse]]></dc:creator>
        <description><![CDATA[
            <p>No no no no no. No not at all.</p>
<p>I’ll try to explain:</p>
<p>As Redash starts, it looks at a setting to know which query runners to enable. Then it loops through the enabled runners and loads their Python dependencies. If a Python <code>ImportError</code> is raised while dependencies are loaded, that query runner is skipped and a log message is emitted. But the process <em>won’t crash</em>, it just skips that query runner and moves along.</p>
<p>However, if a different exception is raised than <code>ImportError</code> this may signal bigger issues with the environment, so non-<code>ImportError</code> exceptions are not explicitly caught and will cause the startup to fail.</p>
<p>Your log messages indicate that startup fails during execution of the line:</p>
<pre data-code-wrap="python3"><code class="lang-nohighlight">import_query_runners(settings.QUERY_RUNNERS)
</code></pre>
<p>And specifically it fails to load dependencies for <code>big_query</code>. The specific dependency that fails is <code>pycrypto</code>. Because of this there are a few ways to debug / correct:</p>
<h3>
<a name="preferred-option-figure-out-why-pycrypto-wont-load-1" class="anchor" href="https://discuss.redash.io#preferred-option-figure-out-why-pycrypto-wont-load-1"></a>[Preferred Option] Figure out why <code>pycrypto</code> won’t load.</h3>
<p>I’m guessing this is a CentOS specific issue (<a href="https://amalgjose.com/2017/11/15/common-dependencies-to-install-pycrypto-package-in-centos-rhel/">context</a>).Once you fix <code>pycrypto</code> for centos then try starting Redash again. This is the preferred option since <code>pycrypto</code> is very widely used across Redash. I believe it is required for the postgres query runner as well, although I’m not sure. I’m away from my system presently so it’s not something I can check.</p>
<h3>
<a name="workaround-1-explicitly-stop-redash-from-trying-to-load-dependencies-for-big-query-bq-using-environment-variableshttpsredashiohelpopen-sourceadmin-guideenv-vars-settings-2" class="anchor" href="https://discuss.redash.io#workaround-1-explicitly-stop-redash-from-trying-to-load-dependencies-for-big-query-bq-using-environment-variableshttpsredashiohelpopen-sourceadmin-guideenv-vars-settings-2"></a>[Workaround 1] Explicitly stop Redash from trying to load dependencies for big query (BQ) using <a href="https://redash.io/help/open-source/admin-guide/env-vars-settings">environment variables</a>.</h3>
<p>The one you’d use to disable big query alone is</p>
<pre><code class="lang-bash">export REDASH_DISABLED_QUERY_RUNNERS="query_runner.big_query"
</code></pre>
<h3>
<a name="workaround-2-tell-redash-to-only-load-dependencies-for-data-sources-you-know-youll-use-3" class="anchor" href="https://discuss.redash.io#workaround-2-tell-redash-to-only-load-dependencies-for-data-sources-you-know-youll-use-3"></a>[Workaround 2] Tell Redash to only load dependencies for data sources you know you’ll use</h3>
<p>If you’re only going to connect with postgres then you can use</p>
<pre><code class="lang-bash">export REDASH_ENABLED_QUERY_RUNNERS="query_runner.pg"
</code></pre>
<hr>
<p>Note that neither workaround will fix this problem if the postgres query runner or one of its dependencies relies on <code>pycrypto</code>. Given there are references in google and stack overflow to special setup instructions with <code>pycrypto</code> on centos, I’d advise fixing the root of the issue.</p>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/10">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/10</link>
        <pubDate>Thu, 16 Jun 2022 13:39:06 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-10</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[dishka_krauch]]></dc:creator>
        <description><![CDATA[
            <p>We use postgres. Do you mean switch redash to on-prem postgres instead of docker container?</p>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/9">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/9</link>
        <pubDate>Thu, 16 Jun 2022 13:15:54 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-9</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[jesse]]></dc:creator>
        <description><![CDATA[
            <p>This looks like an incompatibility between a pinned query runner dependency and your centos installation. Do you know which target databases you hope to query? Such as postgres, mysql etc? If so, I would specify these directly using the enabled_query_runners environment variable. This will prevent Redash trying to load dependencies for query runners you don’t intend to use.</p>
<p>Otherwise you can search here on the forum for centos user posts. There are some eccentricities to centos that have been covered in the past.</p>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/8">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/8</link>
        <pubDate>Thu, 16 Jun 2022 13:01:54 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-8</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[dishka_krauch]]></dc:creator>
        <description><![CDATA[
            <p>No, it’s CentOS7. Follwing your guide here <a href="https://redash.io/help/open-source/dev-guide/docker" rel="noopener nofollow ugc">https://redash.io/help/open-source/dev-guide/docker</a> we faced this pycrypto problem after docker-compose up command.</p>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/7">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/7</link>
        <pubDate>Thu, 16 Jun 2022 12:57:13 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-7</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[jesse]]></dc:creator>
        <description><![CDATA[
            <p>On what target system does this run? A windows machine? macos?</p>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/6">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/6</link>
        <pubDate>Thu, 16 Jun 2022 12:55:04 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-6</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[dishka_krauch]]></dc:creator>
        <description><![CDATA[
            <p>We used docker-compose file from your github page - <a href="https://github.com/getredash/redash/blob/master/docker-compose.yml" class="inline-onebox" rel="noopener nofollow ugc">redash/docker-compose.yml at master · getredash/redash · GitHub</a></p>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/5">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/5</link>
        <pubDate>Thu, 16 Jun 2022 12:53:57 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-5</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[jesse]]></dc:creator>
        <description><![CDATA[
            <p>Redash generally runs with docker compose. Which compose file did you use?</p>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/4">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/4</link>
        <pubDate>Thu, 16 Jun 2022 12:52:41 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-4</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[dishka_krauch]]></dc:creator>
        <description><![CDATA[
            <p>Hey, jesse.<br>
We installed python in Dockerfile with this command:</p>
<p>FROM --platform=linux/amd64 python:3.7-slim-buster</p>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/3">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/3</link>
        <pubDate>Thu, 16 Jun 2022 12:51:35 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-3</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
      <item>
        <title>An error occurs Redash Docker on startup</title>
        <dc:creator><![CDATA[jesse]]></dc:creator>
        <description><![CDATA[
            <p>How did you set up Redash? This looks like a Python version mis-match. Which should not happen if you’ve followed the instructions in our documentation.</p>
          <p><a href="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/2">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608/2</link>
        <pubDate>Tue, 14 Jun 2022 17:30:40 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-10608-2</guid>
        <source url="https://discuss.redash.io/t/an-error-occurs-redash-docker-on-startup/10608.rss">An error occurs Redash Docker on startup</source>
      </item>
  </channel>
</rss>
